resultsdbupdater/utils.py

```diff
@@ -12,20 +12,37 @@
 CONFIG = fedmsg.config.load_config()
 RESULTSDB_API_URL = CONFIG.get('resultsdb-updater.resultsdb_api_url')
 TRUSTED_CA = CONFIG.get('resultsdb-updater.resultsdb_api_ca')
+INVALID_MSG_PUBLISHER_IDS = CONFIG.get(
+    'resultsdb-updater.resultsdb_api_invalid_msg_publisher_ids', ())
 
 LOGGER = logging.getLogger('CIConsumer')
 log_format = '%(asctime)s - %(name)s - %(levelname)s - %(message)s'
 logging.basicConfig(
     format=log_format, level=CONFIG.get('resultsdb-updater.log_level'))
 
 
-def update_publisher_id(data, msg):
+def update_publisher_id(data, msg, testcase):
     """
     Sets data['publisher_id'] to message publisher ID (JMSXUserID) if it
     exists.
+
+    Args:
+        data (dict) - output data for ResultsDB
+        msg (dict) - consumed message
+        testcase (dict) - testcase dictionary for ResultsDB,
+            should contain keys 'name' and 'ref_url'
     """
     msg_publisher_id = msg['headers'].get('JMSXUserID')
-    if msg_publisher_id:
+
+    if msg_publisher_id in INVALID_MSG_PUBLISHER_IDS:
+        msg_id = msg['headers']['message-id']
+        source = testcase.get('ref_url')
+        testcase_name = testcase.get('name')
+        LOGGER.info(
+            'Missing JMSXUserID in message header, '
```

Comment:
This should say "Missing or invalid", right? Also, in the case that it's invalid, it would be nice to log the `msg_publisher_id`.

Reply:
Yes, forgot to change this. Changed to "invalid" and print the value.